ren/tree/nav.hoon

```diff
@@ -0,0 +1,107 @@
+::
+::::  /hoon/nav/tree/ren
+  ::
+/?  310
+/-  tree-include
+/=  gas  /$  fuel:html
+/=  kid-meta  /:  /%%/
+              /^  (map @ta (map knot cord))
+              /_  /%  /tree-front/
+/=  our-meta   /^  (map knot cord)
+               /%  /tree-front/
+/=  bump  /:  /%%/
+          /^  (map @ta @da)
+          /_  /%  /tree-bump/
+::
+=/  sort-opt
+  ?.  (~(has by our-meta) %navsort)
+    'default'
+  (~(got by our-meta) %navsort)
+?>  ?=(@t sort-opt)
+=/  sort-opt  `@tas`(scot %tas sort-opt)
+=/  sorted  %+  sort  ~(tap by kid-meta)
+  |=  $:  [anom=@ta adat=(map knot cord)]
+          [bnom=@ta bdat=(map knot cord)]
+      ==
+  ^-  ?
+  =/  has-a  ~(has by adat)
+  =/  has-b  ~(has by bdat)
+  =/  got-a  ~(got by adat)
+  =/  got-b  ~(got by bdat)
+  ?:  =(%bump sort-opt)
+    ?:  &((~(has by bump) anom) (~(has by bump) bnom))
+      %+  gte  (~(got by bump) anom)
+               (~(got by bump) bnom)
+    ~|  %missing-bump  !!
+  ?:  =(%date sort-opt)
+    ?:  &((has-a %date) (has-b %date))
+      %+  lte  (slav %da (got-a %date)) 
+               (slav %da (got-b %date))
+    ~|  %missing-date  !!  
+  ?:  &((has-a %sort) (has-b %sort))
+    %+  lte  (slav %ud (got-a %sort))
+             (slav %ud (got-b %sort))
+  ?:  (has-a %sort)
+    &
+  ?:  (has-b %sort)
+    |
+  &
+::
+=/  up-link
```

Comment:
These can also be shortened using `++fall`.

Reply:
done